Vert.x Reference Guide - Listening to a Unix Domain Socket - add warning about permissions

### Describe the bug

Vert.x Reference Guide - Listening to a Unix Domain Socket - add warning about permissions 

https://quarkus.io/guides/vertx-reference#listening-to-a-unix-domain-socket 

If I just copy the config snippet
```properties
quarkus.http.domain-socket=/var/run/io.quarkus.app.socket
quarkus.http.domain-socket-enabled=true
```
 and run the dev mode I receive

```
2022-06-02 12:20:32,366 ERROR [io.qua.dep.dev.IsolatedDevModeMain] (main) Failed to recover after failed start: java.lang.RuntimeException: java.lang.RuntimeException: Unable to start HTTP server
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:265)
	at io.quarkus.vertx.http.runtime.devmode.VertxHttpHotReplacementSetup.handleFailedInitialStart(VertxHttpHotReplacementSetup.java:61)
	at io.quarkus.deployment.dev.RuntimeUpdatesProcessor.startupFailed(RuntimeUpdatesProcessor.java:1131)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.firstStart(IsolatedDevModeMain.java:185)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:485)
	at io.quarkus.deployment.dev.IsolatedDevModeMain.accept(IsolatedDevModeMain.java:68)
	at io.quarkus.bootstrap.app.CuratedApplication.runInCl(CuratedApplication.java:142)
	at io.quarkus.bootstrap.app.CuratedApplication.runInAugmentClassLoader(CuratedApplication.java:97)
	at io.quarkus.deployment.dev.DevModeMain.start(DevModeMain.java:132)
	at io.quarkus.deployment.dev.DevModeMain.main(DevModeMain.java:62)
Caused by: java.lang.RuntimeException: Unable to start HTTP server
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:708)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.startServerAfterFailedStart(VertxHttpRecorder.java:257)
	... 9 more
Caused by: java.util.concurrent.ExecutionException: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Permission denied
	at java.base/java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395)
	at java.base/java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1999)
	at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder.java:674)
	... 10 more
Caused by: io.netty.channel.unix.Errors$NativeIoException: bind(..) failed: Permission denied
```

There should be warning, that user needs to have permission to write into `/var/run/io.quarkus.app.socket` location.
The warning should go into docs for sure, would be good to have some info in error message too.

I ended up going with `quarkus.http.domain-socket=/tmp/io.quarkus.app.socket` for experiments :)

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/docs/src/main/asciidoc/vertx-reference.adoc
+++ b/docs/src/main/asciidoc/vertx-reference.adoc
@@ -999,8 +999,8 @@ quarkus.http.so-reuse-port=true
 == Listening to a Unix Domain Socket
 
 Listening on a Unix domain socket allows us to dispense with the overhead of TCP
-if the connection to the quarkus service is established from the same host. This can happen
-if access to the service goes through a proxy which is often the case
+if the connection to the quarkus service is established from the same host.
+This can happen if access to the service goes through a proxy which is often the case
 if you're setting up a service mesh with a proxy like Envoy.
 
 IMPORTANT: This will only work on platforms that support <<native-transport>>.
@@ -1011,6 +1011,8 @@ environment property:
 ----
 quarkus.http.domain-socket=/var/run/io.quarkus.app.socket
 quarkus.http.domain-socket-enabled=true
+
+quarkus.vertx.prefer-native-transport=true
 ----
 
 By itself this will not disable the tcp socket which by default will open on
@@ -1022,6 +1024,10 @@ quarkus.http.host-enabled=false
 
 These properties can be set through Java's `-D` command line parameter or
 on `application.properties`.
+
+IMPORTANT: Do not forget to add the native transport dependency. See <<native-transport>> for details.
+
+IMPORTANT: Make sure your application has the right permissions to write to the socket.
 
 == Read only deployment environments
 

--- a/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
+++ b/extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java
@@ -980,6 +980,15 @@ public class VertxHttpRecorder {
         // Override the host (0.0.0.0 by default) with the configured domain socket.
         options.setHost(httpConfiguration.domainSocket);
 
+        // Check if we can write into the domain socket directory
+        // We can do this check using a blocking API as the execution is done from the main thread (not an I/O thread)
+        File file = new File(httpConfiguration.domainSocket);
+        if (!file.getParentFile().canWrite()) {
+            LOGGER.warnf(
+                    "Unable to write in the domain socket directory (`%s`). Binding to the socket is likely going to fail.",
+                    httpConfiguration.domainSocket);
+        }
+
         return options;
     }
 
@@ -1136,7 +1145,13 @@ public class VertxHttpRecorder {
                         startFuture.complete(null);
                     }
                 } else {
-                    if (event.cause() instanceof IllegalArgumentException) {
+                    if (event.cause() != null && event.cause().getMessage() != null
+                            && event.cause().getMessage().contains("Permission denied")) {
+                        startFuture.fail(new IllegalStateException(
+                                String.format(
+                                        "Unable to bind to Unix domain socket (%s) as the application does not have the permission to write in the directory.",
+                                        domainSocketOptions.getHost())));
+                    } else if (event.cause() instanceof IllegalArgumentException) {
                         startFuture.fail(new IllegalArgumentException(
                                 String.format(
                                         "Unable to bind to Unix domain socket. Consider adding the 'io.netty:%s' dependency. See the Quarkus Vert.x reference guide for more details.",

